dgraphtest/dgraph.go

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright 2023 Dgraph Labs, Inc. and Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dgraphtest
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/docker/docker/api/types/mount"
+	docker "github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/pkg/errors"
+)
+
+const (
+	zeroNameFmt   = "%v_zero%d"
+	zeroLNameFmt  = "zero%d"
+	alphaNameFmt  = "%v_alpha%d"
+	alphaLNameFmt = "alpha%d"
+
+	zeroGrpcPort   = "5080"
+	zeroHttpPort   = "6080"
+	alphaInterPort = "7080"
+	alphaHttpPort  = "8080"
+	alphaGrpcPort  = "9080"
+
+	alphaWorkingDir = "/data/alpha"
+	zeroWorkingDir  = "/data/zero"
+
+	aclSecretPath      = "data/hmac-secret"
+	aclSecretMountPath = "/dgraph-acl/hmac-secret"
+	encKeyPath         = "data/enc-key"
+	encKeyMountPath    = "/dgraph-enc/enc-key"
+)
+
+type dnode interface {
+	cname() string
+	lname() string
+	cid() string
+	ports() nat.PortSet
+	cmd(*Cluster) []string
+	workingDir() string
+	mounts(ClusterConfig) ([]mount.Mount, error)
+	healthURL(c *Cluster) (string, error)
+}
+
+type zero struct {
+	id     int    // 0, 1, 2
+	coid   string // container ID in docker world
```

Comment:
Coid, and coname confused me at first until I saw the struct. What is loname? 
Would it be possible to have friendlier names?

Reply:
done